src/transport/SessionManager.cpp

```diff
@@ -256,13 +256,43 @@ CHIP_ERROR SessionManager::PrepareMessage(const SessionHandle & sessionHandle, P
         return CHIP_ERROR_INTERNAL;
     }
 
-    ChipLogProgress(Inet,
-                    "Prepared %s message %p to 0x" ChipLogFormatX64 " (%u)  of type " ChipLogFormatMessageType
-                    " and protocolId " ChipLogFormatProtocolId " on exchange " ChipLogFormatExchangeId
-                    " with MessageCounter:" ChipLogFormatMessageCounter ".",
-                    sessionHandle->GetSessionTypeString(), &preparedMessage, ChipLogValueX64(destination), fabricIndex,
-                    payloadHeader.GetMessageType(), ChipLogValueProtocolId(payloadHeader.GetProtocolID()),
-                    ChipLogValueExchangeIdFromSentHeader(payloadHeader), packetHeader.GetMessageCounter());
+#if CHIP_PROGRESS_LOGGING
+    CompressedFabricId compressedFabricId = kUndefinedCompressedFabricId;
+
+    if (fabricIndex != kUndefinedFabricIndex && mFabricTable != nullptr)
+    {
+        auto fabricInfo = mFabricTable->FindFabricWithIndex(fabricIndex);
+        if (fabricInfo)
+        {
+            compressedFabricId = fabricInfo->GetCompressedFabricId();
+        }
+    }
+
+    auto * protocolName = Protocols::GetProtocolName(payloadHeader.GetProtocolID());
+    auto * msgTypeName  = Protocols::GetMessageTypeName(payloadHeader.GetProtocolID(), payloadHeader.GetMessageType());
+
+    //
+    // 32-bit value maximum = 10 chars + text preamble (6) + trailer (1) + null (1) + 2 buffer = 20
+    //
+    char ackBuf[20];
+    ackBuf[0] = '\0';
+    if (payloadHeader.GetAckMessageCounter().HasValue())
+    {
+        snprintf(ackBuf, sizeof(ackBuf), " (Ack:" ChipLogFormatMessageCounter ")", payloadHeader.GetAckMessageCounter().Value());
+    }
+
+    //
+    // Legend that can be used to decode this log line can be found in messaging/README.md
+    //
+    ChipLogProgress(ExchangeManager,
+                    "<<< [E:" ChipLogFormatExchangeId " M:" ChipLogFormatMessageCounter "%s] (%s) Msg TX to %u:" ChipLogFormatX64
+                    " [%04X] --- Type %04X:%02X (%s:%s)",
+                    ChipLogValueExchangeIdFromSentHeader(payloadHeader), packetHeader.GetMessageCounter(), ackBuf,
+                    sessionHandle->GetSessionTypeString(), static_cast<uint16_t>(fabricIndex), ChipLogValueX64(destination),
+                    static_cast<uint16_t>(compressedFabricId), static_cast<uint16_t>(payloadHeader.GetProtocolID().GetProtocolId()),
+                    static_cast<uint8_t>(payloadHeader.GetMessageType()), protocolName != nullptr ? protocolName : "---",
```

Comment:
As for the receive end, I don't understand why the casting and nullptr checks are here.